sql: TestRaceWithIndexBackfillMerge failed

sql.TestRaceWithIndexBackfillMerge [failed](https://tanzanite.cluster.engflow.com/invocation/536d5b62-1e8e-4866-ab73-4e5a22917fde) on master @ [22d9482e646ae127981386d77b7ebf729346fc3b](https://github.com/cockroachdb/cockroach/commits/22d9482e646ae127981386d77b7ebf729346fc3b):


```
    mvcc_backfiller_test.go:464: e = 3954, v = 43954
    mvcc_backfiller_test.go:464: e = 3955, v = 43955
    mvcc_backfiller_test.go:464: e = 3956, v = 43956
    mvcc_backfiller_test.go:464: e = 3957, v = 43957
    mvcc_backfiller_test.go:464: e = 3958, v = 43958
    mvcc_backfiller_test.go:464: e = 3959, v = 43959
    mvcc_backfiller_test.go:464: e = 3960, v = 43960
    mvcc_backfiller_test.go:464: e = 3961, v = 43961
    mvcc_backfiller_test.go:464: e = 3962, v = 43962
    mvcc_backfiller_test.go:464: e = 3963, v = 43963
    mvcc_backfiller_test.go:464: e = 3964, v = 43964
    mvcc_backfiller_test.go:464: e = 3965, v = 43965
    mvcc_backfiller_test.go:464: e = 3966, v = 43966
    mvcc_backfiller_test.go:464: e = 3967, v = 43967
    mvcc_backfiller_test.go:464: e = 3968, v = 43968
    mvcc_backfiller_test.go:464: e = 3969, v = 43969
    mvcc_backfiller_test.go:464: e = 3970, v = 43970
    mvcc_backfiller_test.go:464: e = 3971, v = 43971
    mvcc_backfiller_test.go:464: e = 3972, v = 43972
    mvcc_backfiller_test.go:464: e = 3973, v = 43973
    mvcc_backfiller_test.go:464: e = 3974, v = 43974
    mvcc_backfiller_test.go:464: e = 3975, v = 43975
    mvcc_backfiller_test.go:464: e = 3976, v = 43976
    mvcc_backfiller_test.go:464: e = 3977, v = 43977
    mvcc_backfiller_test.go:464: e = 3978, v = 43978
    mvcc_backfiller_test.go:464: e = 3979, v = 43979
    mvcc_backfiller_test.go:464: e = 3980, v = 43980
    mvcc_backfiller_test.go:464: e = 3981, v = 43981
    mvcc_backfiller_test.go:464: e = 3982, v = 43982
    mvcc_backfiller_test.go:464: e = 3983, v = 43983
    mvcc_backfiller_test.go:464: e = 3984, v = 43984
    mvcc_backfiller_test.go:464: e = 3985, v = 43985
    mvcc_backfiller_test.go:464: e = 3986, v = 43986
    mvcc_backfiller_test.go:464: e = 3987, v = 43987
    mvcc_backfiller_test.go:464: e = 3988, v = 43988
    mvcc_backfiller_test.go:464: e = 3989, v = 43989
    mvcc_backfiller_test.go:464: e = 3990, v = 43990
    mvcc_backfiller_test.go:464: e = 3991, v = 43991
    mvcc_backfiller_test.go:464: e = 3992, v = 43992
    mvcc_backfiller_test.go:464: e = 3993, v = 43993
    mvcc_backfiller_test.go:464: e = 3994, v = 43994
    mvcc_backfiller_test.go:464: e = 3995, v = 43995
    mvcc_backfiller_test.go:464: e = 3996, v = 43996
    mvcc_backfiller_test.go:464: e = 3997, v = 43997
    mvcc_backfiller_test.go:464: e = 3998, v = 43998
    mvcc_backfiller_test.go:464: e = 3999, v = 43999
    mvcc_backfiller_test.go:464: e = 4000, v = 44000
    mvcc_backfiller_test.go:477: -- test log scope end --
test logs left over in: outputs.zip/logTestRaceWithIndexBackfillMerge3333129828
--- FAIL: TestRaceWithIndexBackfillMerge (334.65s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRaceWithIndexBackfillMerge.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34291

sql.TestRaceWithIndexBackfillMerge [failed](https://tanzanite.cluster.engflow.com/invocation/c5c25279-aeaf-459c-8ea2-691092f678f7) on master @ [e747f6e6857a19d6048cb184b0f55c52cb8a6390](https://github.com/cockroachdb/cockroach/commits/e747f6e6857a19d6048cb184b0f55c52cb8a6390):


```
    mvcc_backfiller_test.go:464: e = 3954, v = 43953
    mvcc_backfiller_test.go:464: e = 3955, v = 43954
    mvcc_backfiller_test.go:464: e = 3956, v = 43955
    mvcc_backfiller_test.go:464: e = 3957, v = 43956
    mvcc_backfiller_test.go:464: e = 3958, v = 43957
    mvcc_backfiller_test.go:464: e = 3959, v = 43958
    mvcc_backfiller_test.go:464: e = 3960, v = 43959
    mvcc_backfiller_test.go:464: e = 3961, v = 43960
    mvcc_backfiller_test.go:464: e = 3962, v = 43961
    mvcc_backfiller_test.go:464: e = 3963, v = 43962
    mvcc_backfiller_test.go:464: e = 3964, v = 43963
    mvcc_backfiller_test.go:464: e = 3965, v = 43964
    mvcc_backfiller_test.go:464: e = 3966, v = 43965
    mvcc_backfiller_test.go:464: e = 3967, v = 43966
    mvcc_backfiller_test.go:464: e = 3968, v = 43967
    mvcc_backfiller_test.go:464: e = 3969, v = 43968
    mvcc_backfiller_test.go:464: e = 3970, v = 43969
    mvcc_backfiller_test.go:464: e = 3971, v = 43970
    mvcc_backfiller_test.go:464: e = 3972, v = 43971
    mvcc_backfiller_test.go:464: e = 3973, v = 43972
    mvcc_backfiller_test.go:464: e = 3974, v = 43973
    mvcc_backfiller_test.go:464: e = 3975, v = 43974
    mvcc_backfiller_test.go:464: e = 3976, v = 43975
    mvcc_backfiller_test.go:464: e = 3977, v = 43976
    mvcc_backfiller_test.go:464: e = 3978, v = 43977
    mvcc_backfiller_test.go:464: e = 3979, v = 43978
    mvcc_backfiller_test.go:464: e = 3980, v = 43979
    mvcc_backfiller_test.go:464: e = 3981, v = 43980
    mvcc_backfiller_test.go:464: e = 3982, v = 43981
    mvcc_backfiller_test.go:464: e = 3983, v = 43982
    mvcc_backfiller_test.go:464: e = 3984, v = 43983
    mvcc_backfiller_test.go:464: e = 3985, v = 43984
    mvcc_backfiller_test.go:464: e = 3986, v = 43985
    mvcc_backfiller_test.go:464: e = 3987, v = 43986
    mvcc_backfiller_test.go:464: e = 3988, v = 43987
    mvcc_backfiller_test.go:464: e = 3989, v = 43988
    mvcc_backfiller_test.go:464: e = 3990, v = 43989
    mvcc_backfiller_test.go:464: e = 3991, v = 43990
    mvcc_backfiller_test.go:464: e = 3992, v = 43991
    mvcc_backfiller_test.go:464: e = 3993, v = 43992
    mvcc_backfiller_test.go:464: e = 3994, v = 43993
    mvcc_backfiller_test.go:464: e = 3995, v = 43994
    mvcc_backfiller_test.go:464: e = 3996, v = 43995
    mvcc_backfiller_test.go:464: e = 3997, v = 43996
    mvcc_backfiller_test.go:464: e = 3998, v = 43998
    mvcc_backfiller_test.go:464: e = 3999, v = 43999
    mvcc_backfiller_test.go:464: e = 4000, v = 44000
    mvcc_backfiller_test.go:477: -- test log scope end --
test logs left over in: outputs.zip/logTestRaceWithIndexBackfillMerge3764932922
--- FAIL: TestRaceWithIndexBackfillMerge (249.10s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRaceWithIndexBackfillMerge.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/mvcc_backfiller_test.go
+++ b/pkg/sql/mvcc_backfiller_test.go
@@ -273,7 +273,7 @@ func TestRaceWithIndexBackfillMerge(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.UnderStressRace(t, "TODO(ssd) test times outs under race")
+	skip.UnderDuress(t, "TODO(ssd) test times outs under race")
 
 	// protects mergeNotification, writesPopulated
 	var mu syncutil.Mutex

--- a/pkg/testutils/skip/skip.go
+++ b/pkg/testutils/skip/skip.go
@@ -174,6 +174,16 @@ func UnderNonTestBuild(t SkippableTest) {
 
 // UnderDuress skips the test if we are running under any of the
 // conditions we have observed as producing slow builds.
+func UnderDuress(t SkippableTest, args ...interface{}) {
+	t.Helper()
+	if Duress() {
+		skipReason := fmt.Sprintf("duress (current config %s)", testConfig())
+		maybeSkip(t, skipReason, args...)
+	}
+}
+
+// UnderDuressWithIssue skips the test if we are running under any of the
+// conditions we have observed as producing slow builds.
 func UnderDuressWithIssue(t SkippableTest, githubIssueID int, args ...interface{}) {
 	t.Helper()
 	if Duress() {

